pymongoexplain/commands.py

```diff
@@ -39,18 +40,25 @@ def command_name(self):
     def get_SON(self):
         cmd = SON([(self.command_name, self.collection)])
         cmd.update(self.command_document)
+        if self.command_document == {}:
+            return {}
         return cmd
 
 
 class UpdateCommand(BaseCommand):
     def __init__(self, collection: Collection, filter, update,
                  kwargs):
         super().__init__(collection.name)
-        return_document = {"updates":[{"q": filter, "u": update}]}
-        for key in kwargs:
-            value = kwargs[key]
+        return_document = {
+            "updates":[{"q": filter, "u": update}]
+        }
+        for key, value in kwargs.items():
             if key == "bypass_document_validation":
                 return_document[key] = value
+            elif key == "hint":
+                if value is not {} and value is not None:
```

Comment:
We don't need the `{}` check here. It's invalid for hint to be `{}`:
```
>>>> client.t.t.update_one({}, {'$set': {'a':1}}, hint={})
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/shane/git/mongo-python-driver/pymongo/collection.py", line 1024, in update_one
    hint=hint, session=session),
  File "/Users/shane/git/mongo-python-driver/pymongo/collection.py", line 870, in _update_retryable
    _update, session)
  File "/Users/shane/git/mongo-python-driver/pymongo/mongo_client.py", line 1497, in _retryable_write
    return self._retry_with_session(retryable, func, s, None)
  File "/Users/shane/git/mongo-python-driver/pymongo/mongo_client.py", line 1383, in _retry_with_session
    return self._retry_internal(retryable, func, session, bulk)
  File "/Users/shane/git/mongo-python-driver/pymongo/mongo_client.py", line 1415, in _retry_internal
    return func(session, sock_info, retryable)
  File "/Users/shane/git/mongo-python-driver/pymongo/collection.py", line 866, in _update
    retryable_write=retryable_write)
  File "/Users/shane/git/mongo-python-driver/pymongo/collection.py", line 806, in _update
    hint = helpers._index_document(hint)
  File "/Users/shane/git/mongo-python-driver/pymongo/helpers.py", line 87, in _index_document
    "mean %r?" % list(iteritems(index_list)))
TypeError: passing a dict to sort/create_index/hint is not allowed - use a list of tuples instead. did you mean []?
```

Reply:
Done.